### PR TITLE
dbpedia_spotlight

### PR DESCRIPTION
**Pull request by xkgoodbest**
Ke Xu

### CODE REVIEW

File: etk/extractors/dbpedia_spotlight_extractor.py [hunk 1]
```diff
@@ -1,25 +1,66 @@
...
+                           name=extractor_name)
+        self.search_url = url
         # TODO initialise stuff
         # TODO if using the dbpedia webservice, a class variable should be set with the url
         # TODO if using the codebase, the library can be setup to called here
         # TODO if using the codebase, we need a local dbpedia against which the queries are to be made, set it up here as well
-        pass
 
-    def extract(self, text: str) -> List[Extraction]:
+    def extract(self, text: str, confidence: float, filter: list) -> List[Extraction]:
```
Comment (saggu) [inline etk/extractors/dbpedia_spotlight_extractor.py]: confidence and filter should be optional parameters with default value of 0.5 and [Person, Organisation, Place] respectively. `extract` methods in all extractors have the same signature, with optional parameters

File: etk/extractors/dbpedia_spotlight_extractor.py [hunk 2]
```diff
@@ -1,25 +1,66 @@
...
 
 class DBpediaSpotlightExtractor(Extractor):
-    def __init__(self):
+    search_url = ''
+
+    def __init__(self, extractor_name: str, url: str):
+        Extractor.__init__(self, input_type=InputType.TEXT,
+                           category="build_in_extractor",
+                           name=extractor_name)
+        self.search_url = url
```
Comment (saggu) [inline etk/extractors/dbpedia_spotlight_extractor.py]: input parameter should also be called search_url

File: etk/extractors/dbpedia_spotlight_extractor.py [hunk 3]
```diff
@@ -1,25 +1,66 @@
...
+import requests
 
 
 class DBpediaSpotlightExtractor(Extractor):
-    def __init__(self):
+    search_url = ''
+
+    def __init__(self, extractor_name: str, url: str):
+        Extractor.__init__(self, input_type=InputType.TEXT,
+                           category="build_in_extractor",
```
Comment (saggu) [inline etk/extractors/dbpedia_spotlight_extractor.py]: change category to `built_in_extractor`

File: etk/extractors/dbpedia_spotlight_extractor.py [hunk 4]
```diff
@@ -1,25 +1,66 @@
...
             Extract with the input text
             Args:
                 text: str
 
             Returns: List[Extraction]
         """
+
+        self.filter = ','.join(filter)
+
+        # url = 'http://api.dbpedia-spotlight.org/en/annotate'
```
Comment (saggu) [inline etk/extractors/dbpedia_spotlight_extractor.py]: remove all the commented out code

File: etk/extractors/dbpedia_spotlight_extractor.py [hunk 5]
```diff
@@ -1,25 +1,66 @@
...
+            for one_result in resources_results:
+                types = one_result['@types'].split(',')
+                return_result.append(Extraction(confidence=float(results['@confidence']),
+                                                extractor_name=self.name,
+                                                start_char=int(one_result['@offset']),
+                                                end_char=int(one_result['@offset']) + len(one_result['@surfaceForm']),
+                                                value={'surfaceForm': one_result['@surfaceForm'],
+                                                       'URI': one_result['@URI'],
+                                                       'types': types,
+                                                       'similarityScore': float(one_result['@similarityScore'])}))
```
Comment (saggu) [inline etk/extractors/dbpedia_spotlight_extractor.py]: similarityScores -> needs to be changed

File: etk/unit_tests/test_DBPediaSpotlight_extractor.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+            for i in results:
+                extracted_value = {
+                    'value': i.value,
+                    'confidence': i.confidence,
+                    'start_char': i.provenance['start_char'],
+                    'end_char': i.provenance['end_char'],
+                }
+                extracted.append(extracted_value)
+
+            extracted_json = json.dumps(extracted)
```
Comment (saggu) [inline etk/unit_tests/test_DBPediaSpotlight_extractor.py]: remove all the code which you are not using

File: etk/unit_tests/test_DBPediaSpotlight_extractor.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
+import unittest
+import json
+from etk.extractors.dbpedia_spotlight_extractor import DBpediaSpotlightExtractor
+
+
+class TestDBPediaSpotlightExtractor(unittest.TestCase):
```
Comment (saggu) [inline etk/unit_tests/test_DBPediaSpotlight_extractor.py]: change the test cases according to the changes suggested, if needed

**Comment by GreatYYX** [comment]
please also use meaningful commit message

**Comment by xkgoodbest** [comment]
Please check again, I made the changes

**Pull request closed without merging**